src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,45 +1,82 @@
-import React from 'react';
-import { Loader } from '../Loader';
+import React, { useState, useEffect, SetStateAction } from 'react';
+// import { getUsers } from '../../services/getUsers';
+import { User } from '../../types/User';
+import { Todo } from '../../types/Todo';
+import { Loader } from '../Loader/Loader';
+import { getUser } from '../../api';
+
+interface Props {
+  todoCard: Todo | undefined;
+  setIsModalOpen: React.Dispatch<SetStateAction<boolean>>
+  setIconId: React.Dispatch<SetStateAction<number>>
+}
+
+export const TodoModal: React.FC<Props> = ({
+  todoCard,
+  setIsModalOpen,
+  setIconId,
+}) => {
+  const [selectedUser, setsSelectedUser] = useState<User>();
+  const [isUserLoad, setIsUserLoad] = useState(false);
+
+  useEffect(() => {
+    setIsUserLoad(true);
+    getUser(todoCard?.userId).then((user) => {
```

Comment:
it's better to check `if (todoCard?.userId)` and then fetch the user using this function instead of passing `undefined`. It anyway won't return valid info with `undefined`